From a905deed0bf2feb9dc09a5dfa6efbc0b864388d2 Mon Sep 17 00:00:00 2001
From: Konrad Dybcio <konrad.dybcio@linaro.org>
Date: Thu, 16 Feb 2023 12:32:50 +0100
Subject: [PATCH] osm hardcode num freq dom

---
 drivers/cpufreq/qcom-cpufreq-hw.c | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/drivers/cpufreq/qcom-cpufreq-hw.c b/drivers/cpufreq/qcom-cpufreq-hw.c
index 2b22b67b3..a1468d814 100644
--- a/drivers/cpufreq/qcom-cpufreq-hw.c
+++ b/drivers/cpufreq/qcom-cpufreq-hw.c
@@ -237,6 +237,7 @@ struct qcom_cpufreq_hw_params {
 /**
  * struct qcom_cpufreq_soc_data - SoC specific register offsets of the OSM
  *
+ * @num_freq_domains:      Number of frequency domains
  * @reg_enable:            OSM enable status
  * @reg_index:             Index of the Virtual Corner
  * @reg_freq_lut:          Frequency Lookup Table
@@ -250,6 +251,7 @@ struct qcom_cpufreq_hw_params {
  */
 
 struct qcom_cpufreq_soc_data {
+	u8 num_freq_domains;
 	u32 reg_enable;
 	u32 reg_index;
 	u32 reg_domain_state;
@@ -1253,6 +1255,7 @@ static const struct qcom_cpufreq_soc_data qcom_soc_data = {
 };
 
 static const struct qcom_cpufreq_soc_data msm8998_soc_data = {
+	.num_freq_domains = 2,
 	.reg_enable = 0x4,
 	.reg_index = 0x150,
 	.reg_freq_lut = 0x154,
@@ -1749,9 +1752,12 @@ static int qcom_cpufreq_hw_driver_probe(struct platform_device *pdev)
 	if (ret)
 		return dev_err_probe(dev, ret, "Failed to find icc paths\n");
 
-	for (num_domains = 0; num_domains < MAX_FREQ_DOMAINS; num_domains++)
-		if (!platform_get_resource(pdev, IORESOURCE_MEM, num_domains))
-			break;
+	if (soc_data->num_freq_domains)
+		num_domains = soc_data->num_freq_domains;
+	else
+		for (num_domains = 0; num_domains < MAX_FREQ_DOMAINS; num_domains++)
+			if (!platform_get_resource(pdev, IORESOURCE_MEM, num_domains))
+				break;
 
 	qcom_cpufreq.data = devm_kzalloc(dev, sizeof(struct qcom_cpufreq_data) * num_domains,
 					 GFP_KERNEL);
-- 
2.43.0

