diff --git a/drv.c b/drv.c
index b74ed99..1705afa 100644
--- a/drv.c
+++ b/drv.c
@@ -58,6 +58,7 @@ extern const struct backend backend_synaptics;
 extern const struct backend backend_virtgpu;
 extern const struct backend backend_udl;
 extern const struct backend backend_vkms;
+extern const struct backend backend_simpledrm;
 
 extern const struct backend backend_mock;
 
@@ -80,7 +81,7 @@ static const struct backend *drv_backend_list[] = {
 	&backend_evdi,	    &backend_komeda,	&backend_marvell, &backend_mediatek,
 	&backend_meson,	    &backend_nouveau,	&backend_radeon,  &backend_rockchip,
 	&backend_sun4i_drm, &backend_synaptics, &backend_udl,	  &backend_virtgpu,
-	&backend_vkms,	    &backend_mock
+	&backend_vkms,	    &backend_mock, &backend_simpledrm
 };
 
 void drv_preload(bool load)
diff --git a/dumb_driver.c b/dumb_driver.c
index 98c9ba7..9b2eb91 100644
--- a/dumb_driver.c
+++ b/dumb_driver.c
@@ -75,6 +75,7 @@ INIT_DUMB_DRIVER_WITH_NAME(sun4i_drm, "sun4i-drm")
 INIT_DUMB_DRIVER(synaptics)
 INIT_DUMB_DRIVER(udl)
 INIT_DUMB_DRIVER(vkms)
+INIT_DUMB_DRIVER(simpledrm)
 
 #ifndef DRV_ROCKCHIP
 INIT_DUMB_DRIVER(rockchip)
